[ci skip] Active Record Validations guide fixes

jonatack opened the pull request:
- Improve the text explanations for the #add and #full_messages methods
- Remove deprecated `[]=` setter method.
- Fix the #add method that was shown in duplicate. The second code example was originally `[]=`. Replace it with `[] <<`.

kirs committed:
Deprecate alias_method_chain in favour of Module#prepend
…as discussed #19413

Gaurav2728 committed:
remove old unavailable link with relevant fix

jonatack committed:
[ci skip] Consistent spacing inside hash brackets

jonatack committed:
[ci skip] Active Record Validations guide fixes

- Remove deprecated `[]=` method explanation.

- Fix the #add method that was shown in duplicate. The second code example was originally `[]=`, now deprecated. Replace it with `[] <<`.

- Improve the text explanations for the #add and #full_messages methods

yui-knk committed:
[ci skip] Add link for "parameter_names section"

mechanicles committed:
Change 'a' to 'an' for 'HABTM' word [ci skip]

kaspth commented on guides/source/active_record_validations.md:
```diff
@@ -1055,7 +1055,9 @@ person.errors[:name]
 ### `errors.add`
 
-The `add` method lets you manually add messages that are related to particular attributes. You can use the `errors.full_messages` or `errors.to_a` methods to view the messages in the form they might be displayed to a user. Those particular messages get the attribute name prepended (and capitalized). `add` receives the name of the attribute you want to add the message to, and the message itself.
+The `add` method lets you manually add messages that are related to particular attributes. It takes as arguments the name of the attribute receiving the message, and the message itself.
```
I find the explanation for the first argument tough to understand as it's very abstract. Is there a way to make it more concrete?

jonatack replied:
How about "It takes as arguments the name of the attribute and the message."

kaspth commented on guides/source/active_record_validations.md:
```diff
@@ -1055,7 +1055,9 @@ person.errors[:name]
-The `add` method lets you manually add messages that are related to particular attributes. You can use the `errors.full_messages` or `errors.to_a` methods to view the messages in the form they might be displayed to a user. Those particular messages get the attribute name prepended (and capitalized). `add` receives the name of the attribute you want to add the message to, and the message itself.
+The `add` method lets you manually add messages that are related to particular attributes. It takes as arguments the name of the attribute receiving the message, and the message itself.
+
+You can use the `errors.full_messages` or `errors.to_a` methods to view the messages in the form they might be displayed to users, with the capitalized attribute name prepended to the messages.
```
If only they might (versus will), maybe it's not such valuable information to add here? :smile:

jonatack replied:
Thanks for the review! Depends how you write your "display_error_messages" partial (using message or full_messages), right? :smiley: 
Do you prefer "will"?

matthewd replied:
> Do you prefer "will"?

No. "Might" is fine, IMO... but if it needs to be rephrased, something more along the lines of "in a form suitable to be displayed to users" is probably closer.

radanskoric committed:
Fix ActiveModel::Errors#delete return value to stay backward compatible

Rails 5.0 changes to ActiveModel::Errors include addition of `details`
that also accidentally changed the return value of `delete`. Since
there was no test for that behavior it went unnoticed. This commit
adds a test and fixes the regression.

Small improvements to comments have also been made. Since `get` is
getting deprecated it is better to use `[]` in other methods' code
examples. Also, in the module usage example, `def Person.method`
was replaced with a more commonly used `def self.method` code style.

carlosantoniodasilva committed:
Move #19447 changelog to the top and reword it a bit [ci skip]

carlosantoniodasilva committed:
Fix arguments order on assertion

The expected value comes first. Related to #19465.

carlosantoniodasilva committed:
Simplify setting button form options

No need to merge hashes when simply setting options does the job.

jonatack commented on guides/source/active_record_validations.md:
```diff
@@ -1073,12 +1075,12 @@ person.errors.full_messages
 
-Another way to do this is using `[]=` setter
+An equivalent to `errors#add` is to use `<<` to append a message to the `errors.messages` array for an attribute:
 
```
(Fixing lines 1076 and 1081 is the main reason for the PR).

matthewd committed:
Make sure to persist a newly-nil serialized value

The subtype will (quite reasonably) ignore the possibility that it has
`changed_in_place?` by becoming nil.

Fixes #19467

matthewd committed:
Provide a more truthful #inspect

This is obviously all very internal, but sometimes you have to look at
it... and when you do, it'll save a lot of confusion if it doesn't lie
about its identity.

Gaurav2728 committed:
update mysql link that has been replaced

nerdinand committed:
Remove reference to Numeric#from_now, as it is no longer supported

nerdinand committed:
Add documentation for Duration#to_i for clarification

rtomayko committed:
Fix ActionDispatch::PublicExceptions returning string rack status

The status returned in the rack [status, headers, body] array was
a string, which can cause problems with middleware that assumes the
status will be a Fixnum. This likely never surfaced because other
middleware to_i the status returned from downstream apps before
passing it on.

pixeltrix committed:
Fix failing tests for #19474

tgxworld committed:
Remove alias for `i_suck_and_my_tests_are_order_dependent`.

davidcornu committed:
Make sure Array#to_sentence always returns a String

mperham committed:
Pass wrapped class name to Sidekiq for logging purposes

Sidekiq logs the name of the job class being performed.  Because
ActiveJob wraps the class, this means every job logs as an AJ::JobWrapper
instead of the actual class name.

Will help fix mperham/sidekiq#2248

tgxworld committed:
Small doc fix. [CI SKIP]

jonatack committed:
[ci skip] Active Record Validations guide fixes

- Remove the deprecated `[]=`.

- Replace the duplicate errors#add example with `[] <<`.

- Improve the errors#add and errors#full_messages explanations.

jonatack commented:
Will squash.

kaspth commented:
@jonatack Something seems to have gone wrong. There's now 22 commits here.

jonatack commented:
Yes, I've messed up my git history and cannot rebase -i to squash. Closing.

jonatack commented:
Here was the updated text:

The `add` method lets you add an error message related to a particular attribute. It takes as arguments the attribute and the error message.

The `errors.full_messages` method (or its equivalent, `errors.to_a`) returns the error messages in a user-friendly format, with the capitalized attribute

kaspth commented:
Updated text looks great, :+1:

Are you filing a new issue or salvaging the branch by hard resetting to master and cherry-picking your commit?

> Den 24/03/2015 kl. 08.21 skrev Jon Atack <notifications@github.com>:
> 
> Here was the updated text:
> 
> The add method lets you add an error message related to a particular attribute. It takes as arguments the attribute and the error message.
> 
> The errors.full_messages method (or its equivalent, errors.to_a) returns the error messages in a user-friendly format, with the capitalized attribute
> 
> —
> Reply to this email directly or view it on GitHub <https://github.com/rails/rails/pull/19457#issuecomment-85373597>.
> 

--
Kasper

jonatack commented:
Thanks.
Will try the latter, and if no go then the former.
